lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -237,6 +238,33 @@ void VariableOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   setNameFn(getResult(), getName());
 }
 
+llvm::SmallVector<MemorySlot> VariableOp::getPromotableSlots() {
+  if (isa<SVModuleOp>(getOperation()->getParentOp()))
+    return {};
+  return {MemorySlot{getResult(), getType()}};
+}
+
+Value VariableOp::getDefaultValue(const MemorySlot &slot, OpBuilder &builder) {
+  if (auto value = getInitial())
+    return value;
+  return builder.create<ConstantOp>(
+      getLoc(),
+      cast<moore::IntType>(cast<RefType>(slot.elemType).getNestedType()), 0);
+}
+
+void VariableOp::handleBlockArgument(const MemorySlot &slot,
+                                     BlockArgument argument,
+                                     OpBuilder &builder) {}
+
+::std::optional<::mlir::PromotableAllocationOpInterface>
+VariableOp::handlePromotionComplete(const MemorySlot &slot, Value defaultValue,
+                                    OpBuilder &builder) {
+  if (defaultValue.use_empty())
+    defaultValue.getDefiningOp()->erase();
```

Comment:
This is not required, the default value is lazily constructed so if it is unused it will not have been created.